### PR TITLE
chore: usage analytics (BAS only)

### PR DESCRIPTION
**Pull request by bd82**

### CODE REVIEW

**Comment by bd82** [comment]
Still a few TODOs left in the code. 
I need to investigate those.

File: packages/language-server/src/commads.ts [hunk 1]
```diff
@@ -28,6 +29,7 @@ export function executeCommand(
       connection.workspace.applyEdit({
         documentChanges: change,
       });
+      track("MANIFEST_STABLE_ID", "single");
```
Comment (allaVolkov) [inline packages/language-server/src/commads.ts:32]: i think you can use constant TRACK_EVENTS .MANIFEST_STABLE_ID here

File: packages/language-server/src/commads.ts [hunk 2]
```diff
@@ -39,6 +41,7 @@ export function executeCommand(
       connection.workspace.applyEdit({
         documentChanges: change,
       });
+      track("MANIFEST_STABLE_ID", "multiple");
```
Comment (allaVolkov) [inline packages/language-server/src/commads.ts:44]: and here

File: packages/language-server/src/hover.ts [hunk 3]
```diff
@@ -28,6 +29,7 @@ export function getHoverResponse(
   if (astPosition !== undefined) {
     const ui5Node = findUI5HoverNodeAtOffset(astPosition, model);
     if (ui5Node !== undefined) {
+      track("XML_UI5_DOC_HOVER", ui5Node.kind);
```
Comment (allaVolkov) [inline packages/language-server/src/hover.ts:32]: and here

**Review by allaVolkov — APPROVED** [review]
please see minor comments

File: packages/language-server/src/commads.ts (hunk 2 above)
Comment (bd82) [reply packages/language-server/src/commads.ts:44]: I made my own `track` function with a different API.
See:
```typescript
type TrackEventsKeys = keyof typeof TRACK_EVENTS;

export function track(
  eventType: TrackEventsKeys,
  ...custom_events: string[]
): void {
  try {
    getSWA().track(TRACK_EVENTS[eventType], custom_events);
  } catch (e) {
    console.error(`failed usage analytics tracking: ${e.message}`);
  }
}

So it accepts the key for the track event, not the value as the first argument
```

File: packages/language-server/src/hover.ts (hunk 3 above)
Comment (bd82) [reply packages/language-server/src/hover.ts:32]: see answer above

**Review by allaVolkov — APPROVED** [review]

File: packages/language-server/src/hover.ts (hunk 3 above)
Comment (bd82) [reply packages/language-server/src/hover.ts:32]: I must have not posted the answer above...
Anyhow the `track` method I am using here is a different API that accepts the **key** of the relevant event not the value. See:

```typescript
type TrackEventsKeys = keyof typeof TRACK_EVENTS;

export function track(
  eventType: TrackEventsKeys,
  ...custom_events: string[]
): void {
  try {
    getSWA().track(TRACK_EVENTS[eventType], custom_events);
  } catch (e) {
    console.error(`failed usage analytics tracking: ${e.message}`);
  }
}
```

Comment (bd82) [reply packages/language-server/src/hover.ts:32]: oh it was just marked as resolved, never mind :)

**Comment by bd82** [comment]
The coveralls report of -0.2% is related to two tracking calls in a file that is currently not tested at all.
So I cannot resolve it in the scope of this PR.

**Pull request merged**